src/chain_spec.rs

```diff
@@ -112,10 +145,17 @@ where
 }
 
 pub fn centrifuge_config() -> CentrifugeChainSpec {
-	CentrifugeChainSpec::from_json_bytes(
+	let mut spec = CentrifugeChainSpec::from_json_bytes(
 		&include_bytes!("../res/genesis/centrifuge-genesis-spec-raw.json")[..],
 	)
-	.unwrap()
+	.unwrap();
+	let extension = spec.extensions_mut();
+	*extension = Extensions {
```

Comment:
The only thing I am a bit unsure of this approach is that there are values that are not part of the genesis file that are added at runtime. I would love to have static files, that are verifiable, hashable and so on for the chain specs. 
If possible I would regenerate the specs (genesis hash should not change since this is node wrapper stuff) using the regular extension logic. 
We might have to tweak the `$env_staging` functions to regenerate the plain ones and then the raw ones. We can check locally that a node syncs up with the live networks to verify it works.

Reply:
I fully agree! Happy to hear that you think updating the existing chain specs should be feasible. How do we retrieve non-raw specs from our raw specs though? That's not supported by Substrate AFAICT.

Reply:
Updated the chain specs in [a9ba1a7](https://github.com/centrifuge/centrifuge-chain/pull/1392/commits/a9ba1a7f07669d90ed71e0f0c6787bf8ab769cd5) and reverted the `ExtensionsWrapper` in [4d2553f](https://github.com/centrifuge/centrifuge-chain/pull/1392/commits/4d2553fd12674d8ac97e12480fd4ed1f5a3b6946). Unfortunately, our non-raw specs (except for Algol which only exists in human readable format) are so outdated, that building the raw one from these fails with the current client code. They are missing quite a large number of genesis config entries that had been added in the meantime and also contain some properties which do not exist anymore such as `genssis.runtime.system.changesTrieConfig`. Luckily, the extensions properties [are not encoded in the raw specs](https://github.com/paritytech/cumulus/blob/1379520ac152a15dc6f28722538c4ba64449740d/parachains/chain-specs/asset-hub-kusama-genesis.json#L16), so I could just add them there.